src/iface/ethernet.rs

```diff
@@ -666,9 +690,9 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
                 Ok(Packet::None),
 
             _ => {
-                // Send back as much of the original payload as we can
-                let payload_len = cmp::min(
-                    ip_payload.len(), self.device_capabilities.max_transmission_unit);
+                // Send back as much of the original payload as we can.
+                let payload_len = cmp::min(ip_payload.len(),
+                                           IPV4_MIN_MTU - ipv4_repr.buffer_len() * 2 - 8);
```

Comment:
This looks like a bug fix for the existing implementation. Do we need to update existing tests to ensure this behavior stays correct?

Reply:
I already have one for UDP... Not sure if one for this case is necessary